packages/network-console-shared/src/file_io/postman/v2.1/environment-container-adapter.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+    IEnvironmentAdapter,
+    IEnvironmentContainerAdapter,
+    IEnvironmentFormat,
+} from '../../interfaces';
+import {
+    IPostmanEnvironmentFile
+} from './environment-format';
+import { INetConsoleParameter } from '../../../net/net-console-http';
+
+type ValidityCheckResult = {
+    valid: true;
+    parsed: IPostmanEnvironmentFile;
+} | {
+    valid: false;
+    error: string;
+};
+
+export class EnvironmentContainerAdapter implements IEnvironmentContainerAdapter, IEnvironmentAdapter {
+    public readonly canContainMultipleEnvironments = false;
+    private _dirty: boolean;
+    private _current: IPostmanEnvironmentFile;
+    private _childKey: string;
+
+    constructor(
+        public readonly format: IEnvironmentFormat,
+        public readonly id: string,
+        private fileContents: string,
+    ) {
+        const root = EnvironmentContainerAdapter.isValidFile(fileContents);
+        if (!root.valid) {
+            throw new RangeError('Invalid file or error parsing: ' + root.error);
+        }
+
+        this._current = root.parsed;
+        this._dirty = false;
+        this._childKey = `${id}/0`;
+    }
+
+    static isValidFile(fileContents: string): ValidityCheckResult {

```

Comment:
isValidFile and ValidityCheckResult --> parseAndCheck and ParseResult, consistent with the Collection implementation.